lib/user.cpp

```diff
@@ -424,3 +427,15 @@ void User::processEvent(const RoomMemberEvent& event, const Room* room)
         updateAvatarUrl(event.avatarUrl(), d->avatarUrlForRoom(room), room);
     }
 }
+
+qreal User::hueF() const
+{
+    QByteArray hash = QCryptographicHash::hash(d->userId.toUtf8(),
+                                               QCryptographicHash::Sha1);
+    QDataStream dataStream(qToLittleEndian(hash).left(1));
+    dataStream.setByteOrder(QDataStream::LittleEndian);
+    quint8 hashValue;
+    dataStream >> hashValue;
+    qreal hueF = static_cast<qreal>(hashValue)/std::numeric_limits<quint8>::max()*std::numeric_limits<qreal>::max();
```

Comment:
Actually, since we're using XEP-0392 anyway, probably use numbers from [the respective algorithm](https://xmpp.org/extensions/xep-0392.html#algorithm-angle)? Also, it would be great if you add the comment referring to the XEP in the beginning of the method so that others could easily find the source.